libs/src/api/user.js

```diff
@@ -686,6 +691,47 @@ class Users extends Base {
   _cleanUserMetadata (metadata) {
     return pick(metadata, USER_PROPS.concat('user_id'))
   }
+
+  async _updateReplicaSet (userId, metadata) {
+    // Attempt to update through UserReplicaSetManagerClient if present  
+    if (!this.contracts.UserReplicaSetManagerClient) {
```

Comment:
@dmanjunath @SidSethi what happens now is an attempt to initialize this new client and if that fails - fall back to the old UserFactoryClient